sphinx/ext/autosummary/generate.py

```diff
@@ -829,14 +841,19 @@ def main(argv: Sequence[str] = (), /) -> None:
         app.config.templates_path.append(path.abspath(args.templates))
     app.config.autosummary_ignore_module_all = not args.respect_module_all
 
-    generate_autosummary_docs(
+    written_files = generate_autosummary_docs(
         args.source_file,
         args.output_dir,
         '.' + args.suffix,
         imported_members=args.imported_members,
-        app=app,
+        app=app,  # type: ignore[arg-type]
     )
 
+    if args.remove_old:
+        for existing in Path(args.output_dir).glob(f'**/*.{args.suffix}'):
+            if existing not in written_files:
+                existing.unlink()
```

Comment:
I actually missed that one before, but I'd wrap this `unlink()` in case some `OSError` happens and emit a warning.

Reply:
done